### PR TITLE
tray entity added with relationship to vendor entity

### PR DESCRIPTION
**Pull request by atykwonderland**
# Description

added tray entity
should be owned by vendor, vendor can have multiple trays
updated requirements.txt

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.  
Provide instructions so we can **reproduce**.  
Please also list any relevant details for your test configuration

- [x] added a unittest test_vendors_tray() ** make sure to drop all tables before testing **

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I will rebase before merging

### CODE REVIEW

**Comment by yuedai0531** [comment]
to reformat
```
black app/
isort -rc app/
```

**Comment by yuedai0531** [comment]
can you also make sure you rebase to the latest version of master?

**Comment by yuedai0531** [comment]
> can you also make sure you rebase to the latest version of master?

git pull
git rebase origin/master

File: backend/app/main/crud.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
 from sqlalchemy.orm import Session
-
 from . import models, schemas
 
-
 def get_vendor_by_name(session: Session, name: str):
     return session.query(models.Vendor).filter(models.Vendor.name == name).first()
 
+def get_vendor(session: Session, vendor_id: int):
+    return db.query(models.Vendor).filter(models.Vendor.id == vendor_id).first()
```
Comment (taerim25) [inline backend/app/main/crud.py:8]: Shouldn't this be session.query?

File: backend/app/main/crud.py [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+
+def get_vendors(session: Session, skip: int = 0, limit: int = 100):
+    return db.query(models.Vendor).offset(skip).limit(limit).all()
```
Comment (taerim25) [inline backend/app/main/crud.py:11]: Same here

File: backend/app/main/crud.py (hunk 1 above)
Comment (atykwonderland) [reply backend/app/main/crud.py:8]: yesyes it should good catch

File: backend/app/main/crud.py (hunk 2 above)
Comment (atykwonderland) [reply backend/app/main/crud.py:11]: yepyep

**Pull request merged**